platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/dependency_locking.adoc

```diff
@@ -0,0 +1,301 @@
+
+This *lockfile should be included in source control* to ensure that all team members and environments use the exact same dependency versions.
+
+[[migrate_single_lockfile]]
+== Migrating your legacy lockfile
+
+If your project uses the legacy lock file format of a file per locked configuration, follow these instructions to migrate to the new format:
+
+1. Follow the documentation for <<#lock_all_configurations_in_one_build_execution, writing>> or <<#selectively_updating_lock_state_entries, updating>> dependency lock state.
+2. Upon writing the single lock file per project, Gradle will also delete all lock files per configuration for which the state was transferred.
+
+NOTE: Migration can be done one configuration at a time. Gradle will keep sourcing the lock state from the per configuration files as long as there is no information for that configuration in the single lock file.
+
+[[configuring_the_per_project_lock_file_name_and_location]]
+== Configuring the lock file name and location
+
+When using a single lock file per project, you can configure its name and location.
+
+This capability allows you to specify a file name based on project properties, enabling a single project to store different lock states for different execution contexts.
+
+For example, in the JVM ecosystem, the Scala version is often included in artifact coordinates:
+
+====
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingSingleFilePerProject/kotlin",files="build.gradle.kts[tags=locking-file-name]"]
+include::sample[dir="snippets/dependencyManagement/dependencyLocking-lockingSingleFilePerProject/groovy",files="build.gradle[tags=locking-file-name]"]
+====
+
+== Running a build with lock state present
+
+The moment a build needs to resolve a configuration that has locking enabled and it finds a matching lock state,
+it will use it to verify that the given configuration still resolves the same versions.
+
+A successful build indicates that the same dependencies are used as stored in the lock state, regardless if new versions matching the dynamic selector have been produced.
+
+The complete validation is as follows:
+
+* Existing entries in the lock state must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock state
+
+[[fine_tuning_dependency_locking_behaviour_with_lock_mode]]
+=== Fine-tuning dependency locking behaviour with lock mode
+
+While the default lock mode behaves as described above, two other modes are available:
+
+Strict mode::
+In this mode, in addition to the validations above, dependency locking will fail if a configuration marked as _locked_ does not have lock state associated with it.
+
+Lenient mode::
+In this mode, dependency locking will still pin dynamic versions but otherwise changes to the dependency resolution are no longer errors.
```

Comment:
🤔 It would be nice to have an example here.  What other changes are possible?  Adding or removing strictly versioned dependencies?  I don't know, haven't really used this mode.

Reply:
Added.